src/main/java/com/heatheryou/httpserver/route/handler/OptionsHandler.java

```diff
@@ -23,17 +24,17 @@ public Response handle(Request request) {
         return responseBuilder.buildResponse(STATUS_CODE_200, entityHeaders, EMPTY);
     }
 
-    public String getAllowedMethods(Request request) {
+    private String getAllowedMethods(Request request) {
         return EntityHeader.ALLOWED_METHODS + CharacterSet.SPACE + getAllowedMethodsList(request) + CharacterSet.CRLF;
     }
 
-    public String getAllowedMethodsList(Request request) {
+    private String getAllowedMethodsList(Request request) {
         router = new Router();
         List<String> allowedMethods = router.allowedMethods(request.getUri());
         return String.join(",", allowedMethods);
     }
 
-    public String getContentLength() {
+    private String getContentLength() {
```

Comment:
While I always love seeing public surface areas shrink, I think that classes like this benefits strongly from leveraging dependency injection. It currently has an implicit, concrete dependency on `ResponseBuilder` and the methods `getAllowedMethods` and `getContentLength`. While this is a pretty small class, that's a lot of baked in functionality which makes this very resistant to testing at the unit level without setting up several other classes.

Moving functionality like that to abstract types (interfaces) allows them to be injected, which makes this class less dependent upon other components. This lets you provide mocks when unit testing to isolate the class, and it also makes this class much less likely to change if other units do.

For example: if the constructor of `ResponseBuilder` changes to require arguments, this class would have to change. Nothing about how it uses that builder would change, but this code would have to because of the constructor. If it was instead given an already constructed instance in either the handle function or the constructor, it would only need to know what methods to invoke. Changes to the implementation of the builder would not prompt changes here.

(This is a bit of a tempest in a teacup, but it's great to practice in simpler examples as much as possible, since more complex applications can have really big negatives from coupling that get very hard to deal with.)

Reply:
To actually provide an example via C# style naming (because it's better :fist_raised:):

```interface IBuildResponses {
    Response buildResponse(StatusCode statusCode, String[] entityHeaders, String content)
}```

Then in your class here:

```public OptionsHandler(IBuildReponses responseBuilder) {
    this.responseBuilder = responseBuilder;
}```

Then you can easily inject a mock in tests without this class depending on anything else, and additionally, you further define the functionality of your application by creating interfaces that describe responsibilities of your components.

Reply:
This makes a lot of sense, and I appreciate that you took the time to add examples! I still don't have a good habit of thinking about how future changes will impact my current code, so it really helps to see what type of considerations to make to write something that is more "future-proof". This seems like a larger refactoring, so I will make the changes and put them up in the next PR 🙂 and I hope my next project will be in C# so I can see what you mean ✊